### PR TITLE
Add io-openid-provider code review build definition

### PR DESCRIPTION
**Pull request by francescopersico**
### List of changes

- Add io-openid-provider code review build definition.

### Motivation and context

We need the automated part of the code review process to be applied to the new io-openid-provider project.

### Type of changes

- [x] Add new pipeline
- [ ] Update pipeline configuration
- [ ] Remove pipeline

### :warning: If it's new pipeline with code review have you added pagopa-github-bot -> Role: admin?

- [x] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: azure-devops/projects/io-backend-projects/io-openid-provider.tf [hunk 1]
```diff
@@ -0,0 +1,80 @@
+variable "io-openid-provider" {
+  default = {
+    repository = {
+      organization   = "pagopa"
+      name           = "io-openid-provider"
+      branch_name    = "master"
```
Comment (pasqualedevita) [inline azure-devops/projects/io-backend-projects/io-openid-provider.tf:6]:
```suggestion
      branch_name    = "main"
```

File: azure-devops/projects/io-backend-projects/io-openid-provider.tf [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  authorized    = true
+  type          = "endpoint"
+}
+
+# Allow code review pipeline to access Github pr service connection, needed to checkout code from the pull request branch
+resource "azuredevops_resource_authorization" "io-openid-provider-code-review-github-pr-auth" {
+  depends_on = [azuredevops_serviceendpoint_github.io-azure-devops-github-pr, azuredevops_build_definition.io-openid-provider-code-review, azuredevops_project.project]
+
+  project_id    = azuredevops_project.project.id
+  resource_id   = azuredevops_serviceendpoint_github.io-azure-devops-github-pr.id
```
Comment (pasqualedevita) [inline azure-devops/projects/io-backend-projects/io-openid-provider.tf:76]:
```suggestion
  resource_id   = azuredevops_serviceendpoint_github.io-azure-devops-github-rw.id
```
for private repository we need rw permissions

File: azure-devops/projects/io-backend-projects/io-openid-provider.tf [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      }
+    }
+  }
+
+  repository {
+    repo_type             = "GitHub"
+    repo_id               = "${var.io-openid-provider.repository.organization}/${var.io-openid-provider.repository.name}"
+    branch_name           = var.io-openid-provider.repository.branch_name
+    yml_path              = "${var.io-openid-provider.repository.pipelines_path}/code-review-pipelines.yml"
+    service_connection_id = azuredevops_serviceendpoint_github.io-azure-devops-github-pr.id
```
Comment (pasqualedevita) [inline azure-devops/projects/io-backend-projects/io-openid-provider.tf:50]:
```suggestion
    service_connection_id = azuredevops_serviceendpoint_github.io-azure-devops-github-rw.id
```
for private repository we need rw permissions

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: azure-devops/projects/io-backend-projects/io-openid-provider.tf (hunk 2 above)
Comment (francescopersico) [reply azure-devops/projects/io-backend-projects/io-openid-provider.tf:76]: Repository is now public :)

File: azure-devops/projects/io-backend-projects/io-openid-provider.tf (hunk 3 above)
Comment (francescopersico) [reply azure-devops/projects/io-backend-projects/io-openid-provider.tf:50]: Repository is now public :)

**Review by pasqualedevita — APPROVED** [review]

**Comment by pasqualedevita** [comment]
all done :)

**Pull request merged**